### PR TITLE
Add 'secret' code ala chatbox lua tab

### PR DESCRIPTION
**Pull request by ajloveslily14**

### CODE REVIEW

File: lua/gcompute/ui/ide/plugins/metastruct_coh.lua [hunk 1]
```diff
@@ -87,6 +87,7 @@ function self:UpdateHookedView ()
 end
 
 function self:UpdateChatStatus ()
+	local find = string.find
```
Comment (Datamats) [inline lua/gcompute/ui/ide/plugins/metastruct_coh.lua]: why

Comment (ajloveslily14) [reply lua/gcompute/ui/ide/plugins/metastruct_coh.lua]: I suppose I could just have it be string.find(code,"secret",1,true) that seems a bit more cluttered maybe? I'll change it if asked.

File: lua/gcompute/ui/ide/plugins/metastruct_coh.lua [hunk 2]
```diff
@@ -98,6 +99,9 @@ function self:UpdateChatStatus ()
 		if #code > 4096 then
 			code = string.sub (code, 1, GLib.UTF8.GetSequenceStart (code, 4097) - 1) .. "..."
 		end
+		local _, n = code:gsub("\n", '\n')
+		local title = code:match("^[\n%s]*([^\n]+)") or ""
+		code = ( not code:find("secret",1,true) and not code:find("SECRET",1,true)) and code or ("[Coding Lua (%d lines)]\n%s"):format(n, title)
 		coh.SendTypedMessage (code)
```
[the remark this answers is not in the document]
Comment (ajloveslily14) [reply lua/gcompute/ui/ide/plugins/metastruct_coh.lua]: Done.

File: lua/gcompute/ui/ide/plugins/metastruct_coh.lua [hunk 3]
```diff
@@ -98,12 +98,11 @@ function self:UpdateChatStatus ()
...
 			code = string.sub (code, 1, GLib.UTF8.GetSequenceStart (code, 4097) - 1) .. "..."
 		end
-		local _, n = code:gsub("\n", '\n')
-		local title = code:match("^[\n%s]*([^\n]+)") or ""
-		code = ( 
-			not string.find(code,"secret",1,true) and 
-			not string.find(code,"SECRET",1,true)
-		) and code or ("[Coding Lua (%d lines)]\n%s"):format(n, title)
+		local _, n = string.gsub (code, "\n", "\n")
+		local title = string.match (code, "^[\n%s]*([^\n]+)") or ""
```
Comment (notcake) [inline lua/gcompute/ui/ide/plugins/metastruct_coh.lua]: Move this inside the if statement
`%s` covers `\n`
Not all line breaks are `\n`

File: lua/gcompute/ui/ide/plugins/metastruct_coh.lua [hunk 4]
```diff
@@ -98,12 +98,11 @@ function self:UpdateChatStatus ()
 		if #code > 4096 then
 			code = string.sub (code, 1, GLib.UTF8.GetSequenceStart (code, 4097) - 1) .. "..."
 		end
-		local _, n = code:gsub("\n", '\n')
-		local title = code:match("^[\n%s]*([^\n]+)") or ""
-		code = ( 
-			not string.find(code,"secret",1,true) and 
-			not string.find(code,"SECRET",1,true)
-		) and code or ("[Coding Lua (%d lines)]\n%s"):format(n, title)
+		local _, n = string.gsub (code, "\n", "\n")
```
[the remark this answers is not in the document]
Comment (notcake) [reply lua/gcompute/ui/ide/plugins/metastruct_coh.lua]: Rename n to lineCount.
n is off by one.

File: lua/gcompute/ui/ide/plugins/metastruct_coh.lua [hunk 5]
```diff
@@ -98,12 +98,11 @@ function self:UpdateChatStatus ()
...
+		local title = string.match (code, "^[\n%s]*([^\n]+)") or ""
+		if string.find (code, "secret", 1, true) or string.find (code, "SECRET", 1, true) then
```
Comment (notcake) [inline lua/gcompute/ui/ide/plugins/metastruct_coh.lua]: This pattern is too broad, use `-- SECRET` or similar instead.

**Pull request closed without merging**